pkg/appmanager/resourceConfig.go

```diff
@@ -1296,6 +1383,96 @@ func (idg *InternalDataGroup) RemoveRecord(name string) bool {
 	return false
 }
 
+func flattenConflictWarn(key, oldVal, newVal string) string {
+	fmt.Printf("Found mismatch for key '%v' old value: '%v' new value: '%v'\n", key, oldVal, newVal)
+	return oldVal
+}
+
+func flattenConflictConcat(key, oldVal, newVal string) string {
+	// Tokenize both values and add to a map to ensure uniqueness
+	pathMap := make(map[string]bool)
+	for _, token := range strings.Split(oldVal, "|") {
+		pathMap[token] = true
+	}
+	for _, token := range strings.Split(newVal, "|") {
+		pathMap[token] = true
+	}
+
+	// Convert back to an array
+	paths := []string{}
+	for path, _ := range pathMap {
+		paths = append(paths, path)
+	}
+
+	// Sort the paths to have consistent ordering
+	sort.Strings(paths)
+
+	// Write back out to a delimited string
+	var buf bytes.Buffer
+	for i, path := range paths {
+		if i > 0 {
+			buf.WriteString("|")
+		}
+		buf.WriteString(path)
+	}
+
+	return buf.String()
+}
+
+func (dgnm DataGroupNamespaceMap) FlattenNamespaces() *InternalDataGroup {
+
+	// Try to be efficient in these common cases.
+	if len(dgnm) == 0 {
+		// No namespaces.
+		return nil
+	} else if len(dgnm) == 1 {
+		// Only 1 namespace, just return its dg - no flattening needed.
+		for _, dg := range dgnm {
+			return dg
+		}
+	}
+
+	// Use a map to identify duplicates across namespaces
+	var partition, name string
+	flatMap := make(map[string]string)
+	for _, dg := range dgnm {
+		if partition == "" {
+			partition = dg.Partition
+		}
+		if name == "" {
+			name = dg.Name
+		}
+		for _, rec := range dg.Records {
+			item, found := flatMap[rec.Name]
+			if found {
+				if item != rec.Data {
+					conflictFunc, ok := groupFlattenFuncMap[dg.Name]
+					if !ok {
+						log.Warningf("No DataGroup conflict handler defined for '%v'",
+							dg.Name)
+						conflictFunc = flattenConflictWarn
+					}
+					newVal := conflictFunc(rec.Name, item, rec.Data)
+					flatMap[rec.Name] = newVal
+				}
+			} else {
+				flatMap[rec.Name] = rec.Data
+			}
+		}
+	}
+
+	// Create a new datagroup to hold the flattened results
+	newDg := InternalDataGroup{
+		Partition: partition,
+		Name:      name,
+	}
+	for name, data := range flatMap {
+		newDg.AddOrUpdateRecord(name, data)
+	}
+
+	return &newDg
+}
+
```

Comment:
I think it may make more sense to take all of the data group related additions in this file and instead put them into routing.go, where the other data group processing lives. That includes everything in this file except the block in createRSConfigFromRoute where I left an earlier comment.

Reply:
Will do.